tenant/pkg/webhook/default_server/tenantnamespace/validating/tenantnamespace_create_update_handler.go

```diff
@@ -36,19 +41,40 @@ func init() {
 
 // TenantNamespaceCreateUpdateHandler handles TenantNamespace
 type TenantNamespaceCreateUpdateHandler struct {
-	// To use the client, you need to do the following:
-	// - uncomment it
-	// - import sigs.k8s.io/controller-runtime/pkg/client
-	// - uncomment the InjectClient method at the bottom of this file.
-	// Client  client.Client
+	Client client.Client
 
 	// Decoder decodes objects
 	Decoder types.Decoder
 }
 
-func (h *TenantNamespaceCreateUpdateHandler) validatingTenantNamespaceFn(ctx context.Context, obj *tenancyv1alpha1.TenantNamespace) (bool, string, error) {
-	// TODO(user): implement your admission logic
-	return true, "allowed to be admitted", nil
+func (h *TenantNamespaceCreateUpdateHandler) validateTenantNamespaceUpdate(obj *tenancyv1alpha1.TenantNamespace, oldobj *tenancyv1alpha1.TenantNamespace) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if obj.Spec.Name != oldobj.Spec.Name {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec").Child("name"), fmt.Sprintf("cannot modify the name field in spec after initial creation (attempting to change from %s to %s)", oldobj.Spec.Name, obj.Spec.Name)))
+	}
+	return allErrs
+}
+
+func validateTenantNamespaceName(name string, prefix bool) []string {
+	// We don't have name requirement for now
+	return nil
+}
+
+func (h *TenantNamespaceCreateUpdateHandler) validateTenantNamespaceCreate(tList *tenancyv1alpha1.TenantList, obj *tenancyv1alpha1.TenantNamespace) field.ErrorList {
+	path := field.NewPath("metadata")
+	allErrs := apivalidation.ValidateObjectMeta(&obj.ObjectMeta, true, validateTenantNamespaceName, path)
```

Comment:
Is this still required, given that ValidateObjectMetaUpdate wasn't?

Reply:
I leave it here in case we need to put logic in validateTenantNamespaceName later.